packages/jest-circus/src/eventHandler.ts

```diff
@@ -39,9 +39,12 @@ const eventHandler: Circus.EventHandler = (
       const {currentDescribeBlock, currentlyRunningTest} = state;
 
       if (currentlyRunningTest) {
-        throw new Error(
-          `Cannot nest a describe inside a test. Describe block "${blockName}" cannot run because it is nested within "${currentlyRunningTest.name}".`,
+        currentlyRunningTest.errors.push(
+          new Error(
+            `Cannot nest a describe inside a test. Describe block "${blockName}" cannot run because it is nested within "${currentlyRunningTest.name}".`,
+          ),
         );
+        break;
```

Comment:
pushing into the `errors` array is essentially what happened before, except we don't go through the handler again into the `error` case